core-addon/Core.js

```diff
@@ -493,6 +496,23 @@ module.exports = class Core {
       await browser.management.getAll().then(addons =>
         addons.filter(a => a.type == "extension")
           .map(a => a.id));
+
+    // Attempt to resume any paused studies, or pause any running
+    // studies, as appropriate.
+    try {
```

Comment:
Should we wrap the  `this._sendMessageToStudy` in the try instead of the whole iteration loop? For example, this would allow us to keep on pausing/resumng other studies even if a message is thrown for an individual study.